packages/expo-sqlite/android/src/main/cpp/NativeDatabaseBinding.cpp

```diff
@@ -48,6 +50,19 @@ void NativeDatabaseBinding::registerNatives() {
 
 int NativeDatabaseBinding::sqlite3_changes() { return ::exsqlite3_changes(db); }
 
+int NativeDatabaseBinding::sqlite3_finalize_all_statement() {
+  ::exsqlite3_stmt *stmt = ::exsqlite3_next_stmt(db, nullptr); 
+  while (stmt) {
+    ::exsqlite3_stmt *nextStmt = ::exsqlite3_next_stmt(db, stmt);
+    int result = ::exsqlite3_finalize(stmt);
+    if (result != SQLITE_OK) {
+      return result;
+    }
+    stmt = nextStmt;
+  }
+   return SQLITE_OK;
```

Comment:
```suggestion
  ::exsqlite3_stmt *stmt = ::exsqlite3_next_stmt(db, nullptr); 
  int result = SQLITE_OK; 
  while (stmt) {
    ::exsqlite3_stmt *nextStmt = ::exsqlite3_next_stmt(db, stmt);
    int ret = ::exsqlite3_finalize(stmt);
    if (ret != SQLITE_OK) {
      result = ret;
    }
    stmt = nextStmt;
  }
  return result;
```

maybe aggregate the result than having early returns

Reply:
I wasn't sure what was the best practice on this one so thank you for confirming.